backend/stakepoold/rpc/rpcserver/client.go

```diff
@@ -16,15 +16,29 @@ const (
 	disconnectCheckInterval = time.Second * 10
 )
 
+// Client holds the information related to an rpcclient and handles access to
+// that client through a mutex.
 type Client struct {
-	*rpcclient.Client
+	// client is protected by a mutex that must be held for reads/writes.
+	client *rpcclient.Client
+	mux    sync.RWMutex
+
 	cfg          *rpcclient.ConnConfig
 	ntfnHandlers *rpcclient.NotificationHandlers
 	wg           sync.WaitGroup
 	stop         chan struct{}
 	disconnected chan struct{}
 }
 
+// Rpccleint allows access to the underlying rpcclient by providing a copy of
+// its address.
+func (c *Client) Rpcclient() *rpcclient.Client {
```

Comment:
I think a more idiomatic formatting of this function name would be `RPCClient` or `RpcClient`, but I would prefer the former.